buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -668,7 +668,6 @@ class ClusterFormationTasks {
     static Task configureExecTask(String name, Project project, Task setup, NodeInfo node, Object[] execArgs) {
         return project.tasks.create(name: name, type: LoggedExec, dependsOn: setup) { Exec exec ->
             exec.workingDir node.cwd
-            exec.environment 'JAVA_HOME', node.getJavaHome()
```

Comment:
I don't think there is a need to keep java home here. IMO the integration tests should be as true to the expected default setup as they can be. We will have packaging tests to ensure the override behavior works.

Reply:
Note on this that I reversed my previous thinking and added support for RUNTIME_JAVA_HOME. This is because I have experienced a bug with jdk 11.0.2 on macos which prevents Elasticsearch from starting. I haven't been able to get anyone else to replicate this issue, but the support of RUNTIME_JAVA_HOME is the only thing that allows me to run integ tests on my laptop at all with this change.